### PR TITLE
test: Add unit tests for inline-tags utility

### PR DESCRIPTION
**Pull request by hachian**
Added comprehensive unit tests for `src/utils/tag/inline-tags.ts`.
- Covered `generateTagUrl`, `processInlineTags`, and `remarkInlineTags`.
- Verified handling of hierarchical tags, URL encoding, and edge cases.
- Updated `tests/test-runner.js` to include the new test suite.

---
*PR created automatically by Jules for task [14192872781172204927](https://jules.google.com/task/14192872781172204927) started by @hachian*

### CODE REVIEW

File: tests/unit/inline-tags-test.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
+
```
Comment (hachian) [inline tests/unit/inline-tags-test.js]: 先頭に空白があります。

File: tests/unit/inline-tags-test.js [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+import { describe, test } from 'node:test';
+import assert from 'node:assert';
+import { remark } from 'remark';
+import {
+  generateTagUrl,
+  processInlineTags,
+  remarkInlineTags
+} from '../../src/utils/tag/inline-tags.ts';
+
+// Helper: Process markdown with remark-inline-tags
```
Comment (hachian) [inline tests/unit/inline-tags-test.js]: すべてのテスト内容を日本語で説明してください

File: tests/unit/inline-tags-test.js [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+  describe('generateTagUrl', () => {
+    test('基本的なタグURLを生成する', () => {
+      const url = generateTagUrl('test');
+      assert.strictEqual(url, '/tags/test/');
+    });
+
+    test('階層タグURLを生成する（/を-に置換）', () => {
+      const url = generateTagUrl('parent/child');
+      assert.strictEqual(url, '/tags/parent-child/');
```
Comment (hachian) [inline tests/unit/inline-tags-test.js]: 階層タグの記述があっていますか？確認してください  `'/tags/parent/child/'`になるかと思います。

Comment (hachian) [reply tests/unit/inline-tags-test.js]: この時点では`-`が正しく、のちに`/`に変換されるということでしょうか？

Comment (hachian) [reply tests/unit/inline-tags-test.js]: 手元環境で階層タグページにアクセスすると以下のようなURLになります。
http://localhost:4321/tags/tech/web/frontend/

Comment (hachian) [reply tests/unit/inline-tags-test.js]: generateTagUrl 自体にバグがあるのでしょうか。
